refactor: reduce username usage

jns-ps opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Added a new public endpoint to retrieve account information by username, returning the account's unique ID.
  - Registration finalization now returns the new account's unique ID upon success.

- **API Changes**
  - All key, message, and account-related operations now use UUIDs instead of usernames for identification.
  - Message and key endpoints, request/response payloads, and service methods updated to use account IDs (UUIDs).
  - Registration finalization requires an explicit password field in the request body, no longer using the Authorization header.

- **Bug Fixes**
  - Improved error handling for parsing and not-found cases, returning appropriate HTTP status codes.

- **Chores**
  - Updated `.gitignore` to exclude additional IDE directories.
  - Adjusted project run configuration for improved logging and performance.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jns-ps committed:
chore: add .zed directory to gitignore

jns-ps committed:
refactor(account): update account database to return Option<Account>

jns-ps committed:
feat(account): update authentication to use account ID instead of username

jns-ps committed:
feat(account): add account info endpoint to the router

jns-ps committed:
refactor(messages): update message database to use account ID instead of username

jns-ps committed:
refactor(database): update SQLite database to match Option<Account> API

distractedm1nd commented on src/account/database.rs:
```diff
@@ -15,11 +15,11 @@ pub enum AccountDatabaseError {
 pub trait AccountDatabase {
     async fn upsert_account(&self, account: Account) -> Result<(), AccountDatabaseError>;
-    async fn fetch_account(&self, id: Uuid) -> Result<Account, AccountDatabaseError>;
+    async fn fetch_account(&self, id: Uuid) -> Result<Option<Account>, AccountDatabaseError>;
```
nice catch

distractedm1nd reviewed (approved):
very straightforward and clean, good job

jns-ps committed:
chore: simplified Justfile logging env var

jns-ps committed:
refactor(keys): use account ID instead of username for key bundle requests

jns-ps committed:
feat(registration): return account ID when finalizing registration

This is necessary to use the account ID instead of the username for subsequent requests in the client apps.

jns-ps committed:
feat(registration): auth password in request body
